kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiTHttpFrontendService.scala

```diff
@@ -0,0 +1,278 @@
+          info("Thrift HTTP Server SSL: SslContextFactory.getExcludeProtocols = " +
+            String.join(",", sslContextFactory.getExcludeProtocols: _*))
+          sslContextFactory.setKeyStorePath(keyStorePath.get)
+          sslContextFactory.setKeyStorePassword(keyStorePassword.get)
+          new ServerConnector(
+            server.get,
+            sslContextFactory,
+            connectionFactory)
+        } else {
+          new ServerConnector(server.get, connectionFactory)
+        }
+
+      connector.setPort(portNum)
+      // Linux:yes, Windows:no
+      connector.setReuseAddress(!OSUtils.IS_WINDOWS)
+      val maxIdleTime = conf.get(FRONTEND_THRIFT_HTTP_MAX_IDLE_TIME)
+      connector.setIdleTimeout(maxIdleTime)
+      connector.setAcceptQueueSize(maxThreads)
+      server.foreach(_.addConnector(connector))
+
+      val processor = new TCLIService.Processor[TCLIService.Iface](this)
+      val protocolFactory = new TBinaryProtocol.Factory
+      val servlet = new ThriftHttpServlet(processor, protocolFactory, authFactory, conf)
+      servlet.init()
+
+      // Context handler
+      val context = new ServletContextHandler(ServletContextHandler.SESSIONS)
+      context.setContextPath("/")
+
+      context.addEventListener(new ServletContextListener() {
+        override def contextInitialized(servletContextEvent: ServletContextEvent): Unit = {
+          MetricsSystem.tracing { ms =>
+            ms.incCount(THRIFT_CONN_TOTAL)
+            ms.incCount(THRIFT_CONN_OPEN)
+          }
+        }
+
+        override def contextDestroyed(servletContextEvent: ServletContextEvent): Unit = {
+          MetricsSystem.tracing { ms =>
+            ms.decCount(THRIFT_CONN_OPEN)
+          }
+        }
+      })
+
+      val httpPath = getHttpPath(conf.get(FRONTEND_THRIFT_HTTP_PATH))
+
+      if (conf.get(FRONTEND_THRIFT_HTTP_COMPRESSION_ENABLED)) {
+        val gzipHandler = new GzipHandler
+        gzipHandler.setHandler(context)
+        gzipHandler.addIncludedMethods(HttpMethod.POST.asString())
+        gzipHandler.addIncludedMimeTypes(APPLICATION_THRIFT)
+        server.foreach(_.setHandler(gzipHandler))
+      } else {
+        server.foreach(_.setHandler(context))
+      }
+
+      context.addServlet(new ServletHolder(servlet), httpPath)
+      constrainHttpMethods(context)
+
+      info("Started " + getClass.getSimpleName +
+        " in " + schemeName + " mode on port " + portNum +
+        " path=" + httpPath + " with " + minThreads + "..." + maxThreads + " worker threads")
+    } catch {
+      case e: Throwable =>
+        MetricsSystem.tracing(_.incCount(THRIFT_CONN_FAIL))
+        error(e)
+        throw new KyuubiException(
+          s"Failed to initialize frontend service on $serverAddr:$portNum.",
+          e)
+    }
+    super.initialize(conf)
+  }
+
+  override def run(): Unit =
+    try {
+      if (isServer()) {
+        info(s"Starting and exposing JDBC connection at: jdbc:hive2://$connectionUrl/")
```

Comment:
Beeline default use thrift binary mode, we need another parameters for thrift http mode

Reply:
hhtp mode is enabled in beeline/jdbc using the `transportMode=http` option for example `jdbc:hive2://localhost:10010/default;transportMode=http`